grpc-sys/src/lib.rs

```diff
@@ -18,20 +18,38 @@ extern crate libc;
 use libc::{c_char, c_int, c_uint, c_void, size_t, int32_t, int64_t, uint32_t};
 use std::time::Duration;
 
+/// The clocks gRPC supports.
+///
+/// Based on `gpr_clock_type`.
 #[derive(Clone, Copy)]
 #[repr(C)]
 pub enum GprClockType {
+    /// Monotonic clock. Epoch undefined. Always moves forward.
     Monotonic = 0,
+
+    /// Realtime clock. May jump forwards or backwards. Settable by the system administrator.
+    /// Has its epoch at 0:00:00 UTC 1 Jan 1970.
     Realtime,
+
+    /// CPU cycle time obtained by rdtsc instruction on x86 platforms. Epoch undefined. Degrades
+    /// to [`GprClockType::Realtime`] on other platforms.
     Precise,
+
+    /// Unmeasurable clock type: no base, created by taking the difference between two times.
     Timespan,
 }
 
+/// Analogous to struct `timespec`. On some machines, absolute times may be in local time.
+///
+/// Based on `gpr_timespec`.
 #[derive(Clone, Copy)]
 #[repr(C)]
 pub struct GprTimespec {
     pub tv_sec: int64_t,
     pub tv_nsec: int32_t,
+
+    /// Against which clock was this time measured? (or [`GprClockType::Timespan`] if this is a
```

Comment:
It may be more clear to say "Denotes the clock this time was measured from." Asking a question from the reader in documentation can invoke confusion.